### PR TITLE
Oracle Integration testing

### PR DESCRIPTION
**Pull request by tonykploomber**
## Describe your changes

- Add Oracle Integration testing in CI
- Add [Oracle Database doc](https://jupysql--477.org.readthedocs.build/en/477/integrations/oracle.html) in integration part
- Add [Oracle Database part](https://jupysql--477.org.readthedocs.build/en/477/integrations/compatibility.html#oracle-databases) in Compatibility doc

## Issue number

Closes #X

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--477.org.readthedocs.build/en/477/

### CODE REVIEW

**Comment by tonykploomber** [comment]
Not sure why facing the error: `Can't load plugin: sqlalchemy.dialects:oracle.oracledb`

**Comment by tonykploomber** [comment]
@yafimvo 
Hey yafim,

1. Can you help me to test our new Oracle DB container can be running on your windows environment?
2. I am facing some `sqlalchemy.exc.NoSuchModuleError` on our CI, but I don't have this issue on the local?

Once you start the oracle container, you can connect by SQLalchemy by

```
from sqlalchemy import create_engine, text

import oracledb
import pandas as pd
from sql import _testing
engine = create_engine(_testing._get_database_url("oracle"))
engine.connect()

```

to test if the oracle is ready

**Comment by yafimvo** [comment]
@tonykploomber 


> * Can you help me to test our new Oracle DB container can be running on your windows environment?
> * I am facing some `sqlalchemy.exc.NoSuchModuleError` on our CI, but I don't have this issue on the local? Can you check this if possible?

I was able to run this [oracle container](https://hub.docker.com/r/gvenzl/oracle-xe), and use this code to connect to it without errors. tested on Windows 11.

Please note that initially, it takes about ~10 seconds after the container is up to successfully connect to it.

**Comment by tonykploomber** [comment]
@yafimvo 
Do you remember any extra steps you did in the Windows environment to make Oracle run?
If yes, can you help me to update the doc with that?

**Comment by tonykploomber** [comment]
@idomic @yafimvo 
Pardon for the delay due to doc building issue in CI, the related discussion [was here](https://ploomber-internal.slack.com/archives/C03DRTB87HB/p1683267456104559)

The solution is to add `oracle.ipynb` file in `doc/conf.py` in `nb_execution_excludepatterns` section to skip the execution build process

**Review by yafimvo — CHANGES_REQUESTED** [review]
added some notes

File: doc/connecting.md [hunk 1]
```diff
@@ -92,6 +92,7 @@ Check out our guide for connecting to a database:
 - [MSSQL](integrations/mssql)
 - [MySQL](integrations/mysql)
 - [QuestDB](integrations/questdb)
+- [Oracle](integrations/oracle)
```
[the remark this answers is not in the document]
Comment (tonykploomber) [reply doc/connecting.md:95]: Added in https://jupysql--477.org.readthedocs.build/en/477/connecting.html#tutorials

File: doc/integrations/oracle.ipynb [hunk 2]
```diff
@@ -0,0 +1,2443 @@
...
+       "+--------------+-------------+--------------+-------------+------------+"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%sql --with saved_cte SELECT * FROM saved_cte"
```
[the remark this answers is not in the document]
Comment (tonykploomber) [reply doc/integrations/oracle.ipynb]: Done

https://jupysql--477.org.readthedocs.build/en/477/integrations/oracle.html

File: doc/integrations/oracle.ipynb [hunk 3]
```diff
@@ -0,0 +1,2443 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "4ae5475e-5b8f-41a6-936e-8fc07f151ff0",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "threshold = 6.0"
```
[the remark this answers is not in the document]
Comment (tonykploomber) [reply doc/integrations/oracle.ipynb]: It's referring from [Postgresql](https://jupysql--477.org.readthedocs.build/en/477/integrations/postgres-connect.html#parametrize-queries) doc

I guess the purpose is to demonstrate different variable value will display different query result.

But I agree with you, addressed and removed the second threshold part in oracledoc

File: doc/integrations/oracle.ipynb [hunk 4]
```diff
@@ -0,0 +1,2443 @@
...
+   "source": [
+    "%sqlcmd tables"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c7ba0f68-5e44-4879-a051-3ba4e61d0642",
+   "metadata": {},
+   "source": [
+    "List columns in the taxi table:"
```
[the remark this answers is not in the document]
Comment (tonykploomber) [reply doc/integrations/oracle.ipynb]: Done

File: doc/integrations/oracle.ipynb [hunk 5]
```diff
@@ -0,0 +1,2443 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "45cea9dc-84d6-4336-b97b-d07c72e5039e",
+   "metadata": {},
+   "source": [
+    "## Query\n",
+    "\n",
+    "Now, let's start JuppySQL, authenticate and start querying the data!"
```
[the remark this answers is not in the document]
Comment (tonykploomber) [reply doc/integrations/oracle.ipynb]: Good catch, done

File: doc/integrations/oracle.ipynb [hunk 6]
```diff
@@ -0,0 +1,2443 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bab85392-8a67-4831-b501-0aee53c8cc2a",
+   "metadata": {},
+   "source": [
+    "## Load sample data\n",
+    "\n",
+    "Now, let's fetch some sample data. We'll be using the [NYC taxi dataset](https://www.nyc.gov/site/tlc/about/tlc-trip-record-data.page):"
```
[the remark this answers is not in the document]
Comment (tonykploomber) [reply doc/integrations/oracle.ipynb]: Fix taxi -> iris

**Comment by tonykploomber** [comment]
Please review again! Thanks

**Review by yafimvo — APPROVED** [review]
lgtm.

My only concern is that we are using a [non-official oracle image](https://jupysql--477.org.readthedocs.build/en/477/integrations/oracle.html#start-oracle-instance).

@idomic what do you think?

**Comment by edublancas** [comment]
let's see how it goes, if the non-official image gives trouble, we can remove the integration tests and only keep the user guide

**Pull request merged**